src/js/components/search/visualizations/time/TimeVisualizationSection.jsx

```diff
@@ -49,6 +59,64 @@ export default class TimeVisualizationSection extends React.Component {
         }
     }
 
+    downloadLabel = () => {
+        let periodLabel;
+        if (this.props.data.visualizationPeriod === 'fiscal_year') {
+            periodLabel = 'Year';
+        }
+        else {
+            periodLabel = capitalize(this.props.data.visualizationPeriod);
+        }
+        return `Download data by ${periodLabel}`;
+    };
+
+    downloadData = () => {
+        const data = this.props.data;
+        const ret = [];
+        if (data.visualizationPeriod === 'fiscal_year') {
+            ret[0] = ['fiscal_year', 'total_obligations'];
+        }
+        else if (data.visualizationPeriod === 'quarter') {
+            ret[0] = ['fiscal_year', 'fiscal_quarter', 'total_obligations'];
+        }
+        else {
+            ret[0] = ['fiscal_year', 'month', 'total_obligations'];
+        }
+        for (let i = 0; i < data.rawLabels.length; i++) {
+            if (data.visualizationPeriod === 'fiscal_year') {
+                ret[i + 1] = [data.rawLabels[i].year, data.ySeries[i][0]];
+            }
+            else if (data.visualizationPeriod === 'quarter') {
+                ret[i + 1] = [data.rawLabels[i].year, data.rawLabels[i].period[1], data.ySeries[i][0]];
+            }
+            else {
+                const month = data.rawLabels[i].period;
+                ret[i + 1] = [data.rawLabels[i].year, fullMonthFromAbbr(month), data.ySeries[i][0]];
+                if (['Oct', 'Nov', 'Dec'].indexOf(month) > -1) {
+                    // correct FY
+                    ret[i + 1][0] = parseInt(ret[i + 1][0], 10) + 1;
+                }
+            }
+        }
+        return ret;
+    };
+
+    downloadCsv = () => {
+        let contents = this.downloadData();
+        contents = contents.map((row) => row.join(',')).join('\n');
+        // eslint-disable-next-line no-undef
+        const file = new Blob([contents], { type: 'text/csv;charset=utf-8;' });
+        if (isIe()) {
+            window.navigator.msSaveOrOpenBlob(file, 'spending-over-time.csv');
+        }
+        else {
+            const a = document.createElement('a');
```

Comment:
We should just put this in the render, hide it, set the href to a value in state, and set state here. We don't want to avoid using the react conventions. We have all these listeners attached to the document already so we do our best not to interface w/ react.

We should also likely provide a consistent UX for download and present the user with the downloadBottomBar, providing them w/ a link that onClick will download.

We can pair on this and should take like 10min.

Reply:
Yeah, this is unconventional, but since it needs to deal with two radically different approaches (thank you, IE), I don't see how to include both in a different way, but am open to your idea. I don't know what a downloadBottomBar is, but the name sounds misleading in this context.

Reply:
It's a part of our UX. This will have to go through design review and I anticipate they will want a consistent UX for download behavior on this tab. You'll see it and understand if you download a search on the first tab.

Reply:
We should be using react to render this a tag.  the href value should point at state and this should update state.